### PR TITLE
Add config for ironic-image release-24.0

### PR DESCRIPTION
**Pull request by elfosardo**

### CODE REVIEW

File: prow/manifests/overlays/metal3/config.yaml [hunk 1]
```diff
@@ -1532,6 +1539,22 @@ presubmits:
                 value: "TRUE"
             image: docker.io/pipelinecomponents/markdownlint:0.13.0@sha256:9c0cdfb64fd3f1d3bdc5181629b39c2e43b6a52fc9fdc146611e1860845bbae0
             imagePullPolicy: Always
+    - name: markdownlint
```
Comment (tuminoid) [inline prow/manifests/overlays/metal3/config.yaml:1542]: Just add `release-24.0` on L1512, don't copy the config at all.

Comment (elfosardo) [reply prow/manifests/overlays/metal3/config.yaml]: done

File: prow/manifests/overlays/metal3/config.yaml [hunk 2]
```diff
@@ -1487,6 +1494,9 @@ presubmits:
 
   metal3-io/ironic-image:
     - name: shellcheck
+      branches:
```
Comment (tuminoid) [inline prow/manifests/overlays/metal3/config.yaml:1497]: This is not necessary. Shellcheck should be ran for all branches, it has always been there.

Markdownlint we replaced between 23.1 and 24.0, so it differs as we did not backport the patches.

Comment (elfosardo) [reply prow/manifests/overlays/metal3/config.yaml]: weird copy-paste here, I removed it

**Review by tuminoid — COMMENTED** [review]
/approve

**Review by Rozzii — APPROVED** [review]
lgtm

**Review by kashifest — COMMENTED** [review]
/lgtm

we might need to add this branch to whitelist branches in gerrit @tuminoid

**Comment by tuminoid** [comment]
> /lgtm
> 
> we might need to add this branch to whitelist branches in gerrit @tuminoid

release-23.1 is not mentioned in any JJB.

**Comment by kashifest** [comment]
> release-23.1 is not mentioned in any JJB.

Ok then we are good

**Pull request merged**